Add 347 release notes (#6064)
prestosql/presto: presto-docs/src/main/sphinx/release/release-347.md

# Release 347 (XX Nov 2020)

## General Changes

> * Add `ALTER VIEW ... SET AUTHORIZATION` syntax to allow changing the view owner. ({issue}`5789`)

Reviewer: ```suggestion
* Add `ALTER VIEW ... SET AUTHORIZATION` syntax for changing owner of the view. ({issue}`5789`)
```

> * Add support for `ALL` qualifier for `INTERSECT` and `EXCEPT`. ({issue}`2152`)

Author: Yes, but that's not entirely accurate. The language feature is EXCEPT and INTERSECT, which we already support. `ALL` and `DISTINCT` are just capabilities of those features.

> * Add support for `ALL` qualifier for `INTERSECT` and `EXCEPT`. ({issue}`2152`)

Reviewer 2: @martint i see your point. I am trying to make this easier to understand while not incorrect. `INTERSECT ALL` did not work, so it seems fair to say we're adding support for it. up to you

> * Add support for `ALL` qualifier for `INTERSECT` and `EXCEPT`. ({issue}`2152`)

Author: Ok, changed it per your suggestion.

* Add {func}`contains_sequence` function. ({issue}`5593`)

> * Add support for configuring subnet based network topology for topology-aware scheduling. ({issue}`4862`) 

Reviewer: ```suggestion
* Support defining cluster topology (used for query scheduling) using network subnets. ({issue}`4862`) 
```

* Improve query performance by reducing worker to worker communication overhead. ({issue}`5905`, {issue}`5949`)

> * Improve result throughput over fast network links by making http response compression configurable. Clients

Reviewer 3: Maybe rephase:
```
Allow to disable client HTTP results compression, which can improve throughput over fast network links.
Compression can be disabled in CLI using the `--disable-compression` flag.
In JDBC driver it can be disabled using the `disableCompression` driver property.
```

Author: I rephrased it a bit based on your suggestion.

>   can disable compressed responses using the `--disable-compression` flag or `disableCompression` driver property.

Reviewer: what is "driver Property" ?

Author: It refers to the JDBC driver

  Compression can be disabled server-wide with `query-results.compression-enabled=false`. ({issue}`5818`)
* Rename ``rewrite-filtering-semi-join-to-inner-join`` session property to ``rewrite_filtering_semi_join_to_inner_join``. ({issue}`5954`)

> * Throw a user error for failing to decode a session property value. ({issue}`5731`)

Reviewer: ```suggestion
* Throw a user error when session property value can not be decoded. ({issue}`5731`)
```

> * Fix failure when expressions that produce values of type `row` are used in a `VALUES` clause. ({issue}`3398`)

Reviewer: what does "fix failure" mean? Is it "Improve reported error message"?

Reviewer: Oh - the phrasing is good. 
Optionally maybe `Fix query failure` instead `Fix failure`

## Security Changes

* Add support for multiple LDAP bind patterns. ({issue}`5874`)

## Web UI

## JDBC Driver Changes

* Implement `addBatch()`, `clearBatch()` and `executeBatch()` methods in `PreparedStatement`. ({issue}`5507`)

## CLI Changes

> * presto-cli now accepts queries passed via shell redirection. ({issue}`5881`)

Reviewer: ```suggestion
* Add support for passing queries to presto-cli via shell redirection. ({issue}`5881`)
```

Actually I prefer the original phrasing (which I put). But it is not consistent with form we use for other ones.

## Docker Image Changes

> * Update Presto docker image to use Centos 8 as the base. ({issue}`5920`)

Reviewer 3: `as the base.` -> `as the base image`.

## Hive Connector Changes

> * Add support for `ALTER VIEW  ... SET AUTHORIZATION` SQL syntax to change the view owner. This supports Presto and Hive views. ({issue}`5789`)

Reviewer: So is it Hive or general change? Or both?
I guess "both" in terms that parsing logic/analysis needed to be added Presto-wide. But given fact that we have only Hive implementation should we add bullet point to "General" section? In what way changes to the Presto core are user visible?

Author: It's both. There's an item in the General section, as well.

> * Allow configuring HDFS replication factor through property `hive.dfs.replication`. ({issue}`1829`)

Reviewer: ```suggestion
* Allow configuring HDFS replication factor using `hive.dfs.replication` config property. ({issue}`1829`)
```

Author: How about:

```
Allow configuring HDFS replication factor via the `hive.dfs.replication` config property.
```

Reviewer 4: yeah, that sounds good to me too. 👍

> * Add access checks for tables in Hive Procedures. ({issue}`1489`)

Reviewer: Should we list procedures here?

Author: @Praveen2112 ?

> * Add access checks for tables in Hive Procedures. ({issue}`1489`)

Reviewer 2: This is a security bug fix, not "a feature" as it sounds now

Author: It is. As far as I know, it has two parts: 
* Add support in the SPI
* Enforce access control checks within Hive procedures

> * Decrease latency of insert and CTAS queries by updating table and column statistics parallelly. ({issue}`3638`)

Reviewer: ```suggestion
* Decrease latency of insert and CTAS queries by updating table and column statistics in parallel. ({issue}`3638`)
```

* Fix leaking S3 connections when querying Avro tables. ({issue}`5562`)

## Kudu Connector Changes

* Add dynamic filtering support for Kudu. It can be enabled using ``kudu.dynamic-filtering.wait-timeout``
  config property or ``dynamic_filtering_wait_timeout`` session property ({issue}`5594`)

## MongoDB Connector Changes

* Improve performance of queries containing a `LIMIT` clause. ({issue}`5870`)

## SPI Changes

> * Add support for table scan redirections. ({issue}`5792`)

Reviewer: ```suggestion
* Add support for connector to redirect table scan operation to another connector. ({issue}`5792`)
```

> * Add scan physical size stats to query completed event. ({issue}`5872`)

Reviewer: "scan physical size stats" does not read well to me.

Maybe:
```
* Add number of physically scanned bytes to query completed event. ({issue}`5872`)
```

Reviewer 4: yeah, this one reads better to me too. We can say the following to include "rows" as well.

```
* Add the number of physically scanned bytes and rows to query completed event. ({issue}`5872`)
```

Author: How about:

> Add physical input bytes and rows for table scan operation to query completion event.

Reviewer 4: LGTM.

* Allow disabling client HTTP response compression, which can improve throughput over fast network links.
  Compression can be disabled globally via the `query-results.compression-enabled` config property, for CLI via 
  the `--disable-compression` flag, and for the JDBC driver via the `disableCompression` driver property. ({issue}`5818`)
* Rename ``rewrite-filtering-semi-join-to-inner-join`` session property to ``rewrite_filtering_semi_join_to_inner_join``. ({issue}`5954`)
* Throw a user error when session property value cannot be decoded. ({issue}`5731`)
* Fix query failure when expressions that produce values of type `row` are used in a `VALUES` clause. ({issue}`3398`)

## Security Changes

* Add support for multiple LDAP bind patterns. ({issue}`5874`)
* Include groups for view owner when checking permissions for views. ({issue}`5945`)

## JDBC Driver Changes

* Implement `addBatch()`, `clearBatch()` and `executeBatch()` methods in `PreparedStatement`. ({issue}`5507`)

## CLI Changes

* Add support for providing queries to presto-cli via shell redirection. ({issue}`5881`)

## Docker Image Changes

> * Update Presto docker image to use Centos 8 as the base image. ({issue}`5920`)

Reviewer 4: nit: maybe `CentOS`?

## Hive Connector Changes

* Add support for `ALTER VIEW  ... SET AUTHORIZATION` SQL syntax to change the view owner. This supports Presto and Hive views. ({issue}`5789`)
* Allow configuring HDFS replication factor via the `hive.dfs.replication` config property. ({issue}`1829`)
* Add access checks for tables in Hive Procedures. ({issue}`1489`)
* Decrease latency of `INSERT` and `CREATE TABLE AS ...` queries by updating table and column statistics in parallel. ({issue}`3638`)
* Fix leaking S3 connections when querying Avro tables. ({issue}`5562`)

## Kudu Connector Changes

> * Add dynamic filtering support for Kudu. It can be enabled using ``kudu.dynamic-filtering.wait-timeout``

Reviewer 4: Optional suggestion: `Add dynamic filtering support` should be enough as this is already in Kudu section. 

The property seems to be a `Duration`, and the feature is enabled by "setting a non-zero value" for the property. Current sentence structure makes it sound like it's a boolean to me.

```
Add dynamic filtering support. It can be enabled by setting a non-zero duration value for ``kudu.dynamic-filtering.wait-timeout`` config property or ``dynamic_filtering_wait_timeout`` session property.
```
